### PR TITLE
specify user-agent in tmy3 remote request

### LINKED ISSUE #493: NREL TMY server rejecting python requests
**Describe the bug**
NREL's TMY file server recently started rejecting requests originating from urlopen. The response was:

```
"<html><head><title>Request Rejected</title></head><body>An error has occurred. Please consult with your administrator.<br><br>Your support ID is: 12437308730316847769<br><br><a href='javascript:history.back();'>[Go Back]</a></body></html>"
```

However, it's still possible to download a file from the same url with a web browser. Mostly likely they, intentionally or not, started blocking any requests that don't come with a browser-like HTTP User-Agent header.

I have a fix mostly ready to go.

**To Reproduce**
```python
pvlib/test/test_tmy.py F
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

    @network
    def test_readtmy3_remote():
        url = 'http://rredc.nrel.gov/solar/old_data/nsrdb/1991-2005/data/tmy3/703165TYA.CSV'
>       tmy.readtmy3(url)

pvlib/test/test_tmy.py:19:

        # read in file metadata
        meta = dict(zip(head, csvdata.readline().rstrip('\n').split(",")))

        # convert metadata strings to numeric types
>       meta['altitude'] = float(meta['altitude'])
E       KeyError: 'altitude'

pvlib/tmy.py:178: KeyError
```

**Expected behavior**
Should work like it used to!

**Versions:**
all

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #493 
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

### CODE REVIEW

**Comment by thunderfish24** [comment]
@wholmgren Thanks for tracking this one down.

**Comment by wholmgren** [comment]
@cwhanse any concerns here? if not, please go ahead and "squash and merge".

File: pvlib/tmy.py [hunk 1]
```diff
@@ -164,14 +164,23 @@ def readtmy3(filename=None, coerce_year=None, recolumn=True):
 
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
 
-    try:
-        csvdata = open(filename, 'r')
-    except IOError:
-        response = urlopen(filename)
+    if filename.startswith('http'):
+        request = Request(filename, headers={'User-Agent':
+            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) '
```
Comment (cwhanse) [inline pvlib/tmy.py]: Does _any_ user agent work here, or does it need to match the user's browser and platform? If the latter then we need to test this fix on multiple OS.

Comment (wholmgren) [reply pvlib/tmy.py]: No and no. 

By default, Python's urllib specifies a user-agent that is "Python-urllib/x.y" where x.y is the version. https://docs.python.org/3/howto/urllib2.html This user agent is no longer allowed. I am guessing most user agents that identify themselves as scripts are now blocked.

The automated tests run on linux and windows (not mac), and they pass on this PR (note that they currently fail on the master). I don't think that the server has any way of knowing that our script is pretending to be a browser unless it does something smarter (e.g. sees that the requests are coming from IPs associated with travis-ci and appveyor).

Comment (cwhanse) [reply pvlib/tmy.py]: Satisfies me.

**Pull request merged**